src/mscorlib/src/System/Text/Encoding.cs

```diff
@@ -437,92 +466,34 @@ public static Encoding GetEncoding(int codepage)
                     }
 
                     // Double check that we don't have one in the table (in case another thread beat us here)
-                    if ((result = (Encoding)encodings[codepage]) != null)
+                    if ((result = (Encoding)encodings[key]) != null)
```

Comment:
Note: If you decide to follow my advice and use the generic Dictionary instead, then this should be converted to a `TryGetValue` check.